### PR TITLE
studentCSVCRUD frontend

### PR DESCRIPTION
**Pull request by kokinarimoto**
This pull request contains the frontend of the studentCSVCRUD story. There is also some backend code added to this; however, the full code will be for the backend of the studentCSVCRUD story.

We have rated this a 4 due to the various components needed to implement a fully functional CRUD for students and to be able to implement adequate test coverage for the different cases.

### CODE REVIEW

**Review by gracsoo — APPROVED** [review]
everything looks good :)

**Review by mara-downing — CHANGES_REQUESTED** [review]
Preliminary concern: I'm uploading a file that just looks like:
`
mail,team
student1@email,team1
student2@email,team2
`
and it doesn't seem to upload, the error in the console looks like:
Screenshot from 2020-12-11 20-28-38

Let me know if I'm doing something wrong with the input file, otherwise this would be an error to take a look into.

File: javascript/src/main/utils/fetch.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ export async function fetchWithToken(url, getToken, options) {
   });
   if (response.status >= 400 && response.status < 600) {
     throw new Error(response.error_description);
+    // const json = await response.json();
+    // throw new Error("HTTP Error " + json.status + ": " + json.message);
```
Comment (phtcon) [inline javascript/src/main/utils/fetch.js:15]: The usual practice is to delete commented out code before doing a PR.

Could we delete these lines?

File: javascript/src/test/pages/Students/EditStudent.test.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      data: undefined,
+      error: undefined,
+      mutate: mutateSpy,
+    });
+    const { getByAltText } = render(<EditStudent />);
+    const loading = getByAltText("Loading");
+    expect(loading).toBeInTheDocument();
+  });
+
+  // test("Renders spinner with no existing student", () => {
```
Comment (phtcon) [inline javascript/src/test/pages/Students/EditStudent.test.js:91]: Could we delete these commented out lines?

File: javascript/src/test/pages/Students/EditStudent.test.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    const { getByText } = render(
+      <EditStudent />
+    );
+
+    const submitButton = getByText("Submit");
+    expect(submitButton).toBeInTheDocument();
+    userEvent.click(submitButton);
+
+    // expect(addToast).toHaveBeenCalledTimes(1);
```
Comment (phtcon) [inline javascript/src/test/pages/Students/EditStudent.test.js:154]: Could we delete these commented out lines?

File: javascript/src/test/pages/Students/NewStudent.test.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+import { fetchWithToken } from "main/utils/fetch";
+jest.mock("main/utils/fetch", () => ({
+  fetchWithToken: jest.fn()
+}));
+
+// import { useToasts } from 'react-toast-notifications'
+// jest.mock('react-toast-notifications', () => ({
+//   useToasts: jest.fn()
+// }));
```
Comment (phtcon) [inline javascript/src/test/pages/Students/NewStudent.test.js:24]: Delete commented out lines

File: javascript/src/test/pages/Students/StudentCSVButton.test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    // test("renders an error message when there is an error", () => {
+    //     useSWR.mockReturnValue({
+    //       data: undefined,
+    //       error: new Error("this is an error"),
+    //       mutate: mutateSpy,
+    //     });
+    //     const { getByText } = render(<StudentCSVButton />);
+    //     const error = getByText(/error/);
+    //     expect(error).toBeInTheDocument();
+    //   });
```
Comment (phtcon) [inline javascript/src/test/pages/Students/StudentCSVButton.test.js:73]: delete commented out lines

File: javascript/src/test/pages/Students/Students.test.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+  test("can delete a student", async () => {
+    const fakeDeleteFunction = jest.fn();
+    buildDeleteStudent.mockReturnValue(fakeDeleteFunction);
+    // fakeDeleteFunction.mockImplementation(async () => {
+    //   // If you want it to work, do nothing
+
+    //   // Otherwise, throw an error
+    //   throw new Error();
+    // });
```
Comment (phtcon) [inline javascript/src/test/pages/Students/Students.test.js:97]: You guessed it

File: javascript/src/test/pages/Students/Students.test.js [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    const deleteButtons = getAllByTestId("delete-button");
+    userEvent.click(deleteButtons[0]);
+    await waitFor(() => expect(fakeDeleteFunction).toHaveBeenCalledTimes(1));
+  });
+
+  // test("can delete all students", async () => {
+  //   const fakeDeleteFunction = jest.fn();
+  //   buildDeleteStudent.mockReturnValue(fakeDeleteFunction);
+  //   fakeDeleteFunction.mockImplementation(async () => {
+  //     // If you want it to work, do nothing
```
Comment (phtcon) [inline javascript/src/test/pages/Students/Students.test.js:108]: You know what to do

**Review by phtcon — CHANGES_REQUESTED** [review]
Some commented out code to delete

**Comment by phtcon** [comment]
Closed because PR #91 has the code from the branch that the team decided to go forward with

**Pull request closed without merging**